sale_product_approval/models/product_product.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2021 Open Source Integrators
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    sale_ok = fields.Boolean(
+        compute="_compute_sale_ok_product",
+        string="Can be Sold",
+        default=False,
+        store=True,
+    )
+    candidate_sale = fields.Boolean(
+        string="Candidate to be Sold",
+        default=True,
+    )
+
+    @api.depends("product_state_id")
```

Comment:
Missing dependencies
```suggestion
    @api.depends("candidate_sale", "product_state_id.approved_sale")
```